packages/datatrak-web-server/src/routes/EntitiesRoute.ts

```diff
@@ -30,14 +30,20 @@ async function getEntityCodeFromId(services: TupaiaApiClient, id: string) {
 
 export class EntitiesRoute extends Route<EntitiesRequest> {
   public async buildResponse() {
-    const { query, ctx } = this.req;
+    const { query, ctx, models, session } = this.req;
     const { services } = ctx;
+    let recentEntities: string[] = [];
 
     const {
       filter: { countryCode, projectCode, grandparentId, parentId, searchString, type },
       fields = DEFAULT_FIELDS,
     } = query;
 
+    if (!!session) {
+      const currentUser = await models.user.findOne({ email: session.email });
+      recentEntities = currentUser.preferences.recentEntities?.[countryCode]?.[type] || [];
```

Comment:
Will the user record all have `recentEntities`?

Reply:
Not necessarily. That's why we've got the `?.` operator carrying this. Past a certain point newly logged in users will always have `recentEntities` since the empty object is assigned when we assign a `project_id`, but people who have already logged in prior to us adding this behaviour won't have this field until they either submit an entity answer or change projects